chapter_12/2_Mapper_algorithm.tex

```diff
@@ -0,0 +1,16 @@
+%Contributor: Yixuan Li 
+\section{The Mapper algorithm}
+\textbf{Definition 2.} Let $f: X \rightarrow \mathbb{R}^d, d\geq 1$, be a continuous real valued function and let $U = (U_i)_{i\in I}$ be a cover of $\mathbb{R}^d$. The pull back cover of $X$ induced by $(f, U)$ is the collection of open sets $(f^{-1}(U_i))_{i\in I}$. The refined pull back is the collection of connected components of the open sets $f^{-1}(U_i), i \in I$.
+\begin{algorithm}[H]
+\SetAlgoLined
```

Comment:
This uses the algorithm2e package which isnt used in the text. Please change to use the algorithmic package instead.